hermes-test-helper/src/main/java/pl/allegro/tech/hermes/test/helper/endpoint/BrokerOperations.java

```diff
@@ -28,63 +30,73 @@
-
-                                   return new KafkaZkClient(zooKeeperClient, false, Time.SYSTEM);
-                               }));
+    public BrokerOperations(Map<String, String> kafkaConnection, ConfigFactory configFactory) {
+        adminClients = kafkaConnection.entrySet().stream()
+                .collect(toMap(Map.Entry::getKey, e -> brokerAdminClient(e.getValue())));
+        String namespace = configFactory.getStringProperty(Configs.KAFKA_NAMESPACE);
+        String namespaceSeparator = configFactory.getStringProperty(Configs.KAFKA_NAMESPACE_SEPARATOR);
         kafkaNamesMapper = new JsonToAvroMigrationKafkaNamesMapper(namespace, namespaceSeparator);
     }
 
     public void createTopic(String topicName) {
-        zkClients.values().forEach(c -> createTopic(topicName, c));
+        adminClients.values().forEach(c -> createTopic(topicName, c));
     }
 
     public void createTopic(String topicName, String brokerName) {
-        createTopic(topicName, zkClients.get(brokerName));
+        createTopic(topicName, adminClients.get(brokerName));
     }
 
-    private void createTopic(String topicName, KafkaZkClient kafkaZkClient) {
+    private void createTopic(String topicName, AdminClient adminClient) {
         Topic topic = topic(topicName).build();
-        kafkaNamesMapper.toKafkaTopics(topic).forEach(kafkaTopic -> {
-            AdminZkClient adminZkClient = new AdminZkClient(kafkaZkClient);
-            adminZkClient.createTopic(kafkaTopic.name().asString(), DEFAULT_PARTITIONS, DEFAULT_REPLICATION_FACTOR, new Properties(), RackAwareMode.Enforced$.MODULE$);
+        kafkaNamesMapper.toKafkaTopics(topic)
+                .forEach(kafkaTopic -> createTopic(adminClient, kafkaTopic.name().asString()));
+    }
 
-            waitAtMost(adjust(Duration.ONE_MINUTE)).until(() -> {
-                        kafkaZkClient.topicExists(kafkaTopic.name().asString());
-                    }
-            );
-        });
+    private void createTopic(AdminClient adminClient, String topicName) {
+        try {
+            NewTopic topic = new NewTopic(topicName, DEFAULT_PARTITIONS, (short) DEFAULT_REPLICATION_FACTOR);
+            adminClient.createTopics(singletonList(topic))
+                    .all()
+                    .get(1, MINUTES);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
     }
 
     public boolean topicExists(String topicName, String kafkaClusterName) {
         Topic topic = topic(topicName).build();
         return kafkaNamesMapper.toKafkaTopics(topic)
-                .allMatch(kafkaTopic -> zkClients.get(kafkaClusterName).topicExists(kafkaTopic.name().asString()) &&
-                        !isMarkedForDeletion(kafkaClusterName, kafkaTopic));
+                .allMatch(kafkaTopic -> topicExists(kafkaClusterName, kafkaTopic));
+    }
+
+    private boolean topicExists(String kafkaClusterName, KafkaTopic kafkaTopic) {
+        try {
+            adminClients.get(kafkaClusterName)
```

Comment:
I think that this should also work :)
```java
return adminClients.get(kafkaClusterName)
        .listTopics()
        .names()
        .get(1, MINUTES)
        .contains(kafkaTopic.name().asString());
```

Also, this might reduce `try/catch` logic determining whether topic exists or not.

Reply:
good point, fixed